pymc/variational/approximations.py

```diff
@@ -124,16 +121,7 @@ def __init_group__(self, group):
         self._finalize_init()
 
     def create_shared_params(self, start=None):
-        if start is None:
-            start = self.model.initial_point
-        else:
-            start_ = start.copy()
-            self.model.update_start_vals(start_, self.model.initial_point)
-            start = start_
-        if self.batched:
-            start = start[self.group[0].name][0]
-        else:
-            start = DictToArrayBijection.map(start)
+        start = self._prepare_start(start)
```

Comment:
Can you specify type hints for the `start` kwarg?
In #4983 I also needed to edit this file, so one of us will get a git conflict. Please check what I did over there and let's discuss which PR should be merged first?